sqllineage/core/parser/sqlfluff/extractors/dml_merge_extractor.py

```diff
@@ -110,13 +111,15 @@ def extract(
                     tgt_col.parent = list(holder.write)[0]
                     insert_columns.append(tgt_col)
                 for j, e in enumerate(
-                    merge_insert.get_child("values_clause")
-                    .get_child("bracketed")
-                    .get_children("expression")
+                    retrieve_segments(
+                        merge_insert.get_child("values_clause").get_child("bracketed"),
+                        check_bracketed=False,
+                    )
                 ):
-                    col_ref = e.get_child("column_reference")
-                    if col_ref:
-                        src_col = Column(get_identifier(col_ref))
-                        src_col.parent = direct_source
-                        holder.add_column_lineage(src_col, insert_columns[j])
+                    if e.type == "expression":
+                        col_ref = e.get_child("column_reference")
+                        if col_ref:
+                            src_col = Column(get_identifier(col_ref))
+                            src_col.parent = direct_source
+                            holder.add_column_lineage(src_col, insert_columns[j])
```

Comment:
This change is required as with `sqlfluff==2.1.4` literal values are considered as type literal and not column expression anymore